rosidl_typesupport_opensplice_cpp/rosidl_typesupport_opensplice_cpp/__init__.py

```diff
@@ -24,6 +25,28 @@
 from rosidl_parser import validate_field_types
 
 
+def check_idlpp_supports_include_namespaces(idl_pp):
+    res = 0
+
+    r = subprocess.getstatusoutput(idl_pp + ' -v')
+    ospl_version = r[1].split(':')[1].strip()
+
+    if ospl_version.endswith('OSS'):
+        m = re.search('([1-9][0-9]*\.[0-9]+)\.([0-9]*)', ospl_version)
+        if m and m.lastindex == 2:
+            major = m.group(1)
+            minor = m.group(2)
+            res = (major > '6.9' or (major == '6.9' and minor > '181127'))
+    else:
+        m = re.search('([1-9][0-9]*\.[0-9]+)\.([0-9]*)', ospl_version)
+        if m and m.lastindex == 2:
+            major = m.group(1)
+            minor = m.group(2)
+            res = (major > '6.10' or (major == '6.10' and minor > '1'))
```

Comment:
It looks like that the majority of the two blocks are identical. Please move the generic part above the condition and only make the version number comparison line conditional.